xtask/tests/tidy.rs

```diff
@@ -52,15 +52,36 @@ fn rust_files_are_tidy() {
 
 #[test]
 fn check_merge_commits() {
-    let cmd_output =
-        run!("git rev-list --merges --invert-grep --author 'bors\\[bot\\]' HEAD~4.."; echo = false);
-    match cmd_output {
-        Ok(out) => {
-            if !out.is_empty() {
-                panic!("Please rebase your branch on top of master by running `git rebase master`");
-            }
-        }
-        Err(e) => panic!("{}", e),
+    let stdout = run!("git rev-list --merges --invert-grep --author 'bors\\[bot\\]' HEAD~19.."; echo = false)
+        .unwrap();
+    if !stdout.is_empty() {
+        panic!("\
+
+Merge commits are not allowed in the history.
+
+When updating a pull-request, please rebase your feature branch
+on top of master by running `git rebase master`. If rebase fails,
+you can re-apply your changes like this:
+
+  # Abort in-progress rebase, if any.
+  $ git rebase --abort
+
+  # Make the branch point to the latest commit from master,
+  # while maintaining your local changes uncommited.
+  $ git reset --soft origin/master
+
+  # Commit all changes in a single batch.
+  $ git commit -am'My changes'
```

Comment:
```suggestion
  $ git commit -am 'My changes'
```

Reply:
Hm, I personally never type a space here, and it works

Reply:
Surprising. `-am'My changes'` and `'-amMy changes'` are passed the same on Unix. Most programs would interpret it as `-a -m -M -y '- ' -c -h -a -n -g -e -s`. There seems to be a special case for `-m` in `git commit`.

Reply:
Wow!

I wonder if might be the behavior of my shell (fish)?

Reply:
It also works fine with my shell (zsh)